go/vt/vtgate/vtgate.go

```diff
@@ -221,6 +224,10 @@ func Init(ctx context.Context, hc discovery.HealthCheck, topoServer *topo.Server
 	}
 
 	errorCounts = stats.NewMultiCounters("VtgateApiErrorCounts", []string{"Operation", "Keyspace", "DbType", "Code"})
+	promErrorCounts = promstats.NewCounter(
```

Comment:
I don't think that defining each metric twice is the way to go here...
Instead my thought was that you would do something like this:

```
errorCounts = stats.NewMultiCounters(
    "VtgateApiErrorCounts",
    "api_error_counts",
    "Vtgate API error counts per error type",
    []string{"Operation", "Keyspace", "DbType", "Code"}
)
```

In other words when creating the abstraction you give it all the information needed to create the underlying prom counter.

Then inside the stats implementation you would crate all the necessary hooks to make the corresponding prom metric object.

Reply:
Yeah, this is what I was wondering about approach-wise. That fits with what I was thinking you'd think. : ) 

My one concern is how much extra stuff we might need to pass through for rates (presuming we want to support them prom-native), but I"ll take a look at a rates example to see what we need to do.